### ISSUE TITLE
Add Project Treble compatible phones.

### ISSUE THREAD

**Issue by dch82** [openandroidinstaller-dev/openandroidinstaller#156]
For those who don't know, Project Treble (also known as GSI) allows any compatible phone (i.e. any phone that is bootloader-unlocked, is launched with Android 9 or above, and has GMS).
This will widen the reach by a lot.
BTW, you can find more information on the Android dev website.

**Comment by tsterbak** [comment]
Thanks for the suggestion! :slightly_smiling_face: 

Can you provide a bit more information about the project? I find mostly old stuff and nothing on recent phones and how it is actually installed.

**Comment by dch82** [comment]
I use a bootloader-unlocked Nokia 8.3. This was how I installed (with only `fastboot` and `adb`):

***This may brick your device. Some other devices did get bricked***
1. Unlock USB debugging.
2. Run `adb reboot fastboot`
3. After it is done, `fastboot reboot fastboot` to get into `fastbootd`
4. If your phone has a dedicated `vbmeta` partition, run `fastboot flash vbmeta --disable-verity --disable-verification vbmeta.img`
5. `fastboot erase system` to remove preexisting os
6. `fastboot flash system system.img` to flash system
7. There might be an error saying something like not enough space available. If so, run `fastboot delete-logical-partition product_a`

Notably, I left out bootloader unlocking, because it is different for every phone, and you can obtain `vbmeta.img` from any Google official gsi download, `system.img` from any gsi rom.

**Comment by dch82** [comment]
> Thanks for the suggestion! :slightly_smiling_face: 
> 
> Can you provide a bit more information about the project? I find mostly old stuff and nothing on recent phones and how it is actually installed.

OK, have you have ever used an un-modded android phone, and waited for a shiny new version of Android to reach your phone?

If you weren't using Android One, Google Pixel, or some other speedy manufacturer, you could have to wait for quite a long time for the new version to land on your phone.

Google knew about this, and decided to create a new standardized android base called Project Treble to standardize ~~all the~~ most of the different android blobs and drivers into one image, which the manufacturers can modify quickly as, for example, Motorola doesn't need to wait for its SOC manufacturers to create a derivative image for a new SOC to start development; the SOC manufacturers will upstream their work onto the GSI.

Because the most essential android blobs and drivers are on one image, there is a side effect that any treble-compatible phone can run any treble-compatible image/rom!

In other words, GSI allows modders to customize almost any bootloader-unlocked newer android phone to their liking.
For manufacturers, it promises rapid schedules and a decluttered tree of images; For modders, it increases the amount of moddable phones available

**Comment by SirRGB** [comment]
Ik this is quite an old issue, but still open, so here is my 5 cents...
I personally think that treble works nicely for some devices and pretty badly for others. It can be confusing for users if stuff just doesnt work even if they correctly applied all steps. This gets even worse considering there are GSIs, that apply lots of hacks for maximum compatibility and others, that dont. So issues can differ on the same phone and Android version on the same phone. In my opinion openandroidinstaller is for less techy people, who dont want to troubleshoot or cant when their phone doesnt boot.
I know you recommend to use A9+ devices, but even now there is some oems, that just dont comply with treble...

https://github.com/phhusson/treble_experimentations/wiki

**Comment by dch82** [comment]
Ok, I'll be closing this for now.
